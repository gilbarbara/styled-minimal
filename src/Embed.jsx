import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Box from './Box';

const StyledEmbed = styled(Box)`
  overflow: hidden;
  position: relative;
  width: 100%;
  
  & > iframe {
    border: 0;
    bottom: 0;
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    width: 100%;
  }
`;

const Embed = ({ children, ...props }) => (
  <StyledEmbed dangerouslySetInnerHTML={{ __html: children }} {...props} />
);

Embed.propTypes = {
  children: PropTypes.node.isRequired,
  ratio: PropTypes.number,
  ...Box.propTypes,
};

Embed.defaultProps = {
  ratio: 9 / 16,
};

export default Embed;
