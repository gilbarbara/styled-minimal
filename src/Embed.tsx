import * as React from 'react';
import styled, { css } from 'styled-components';

import { baseStyles, sx, system } from './utils/system';

import { EmbedProps } from './types';

export const StyledEmbed = styled('div')<Omit<EmbedProps, 'children'>>(
  props => {
    const { ratio } = props;

    return css`
      ${baseStyles};
      height: 0;
      overflow: hidden;
      padding-bottom: ${`${ratio! * 100}%`};
      position: relative;
      width: 100%;

      & > iframe {
        border: 0;
        bottom: 0;
        height: 100%;
        left: 0;
        position: absolute;
        top: 0;
        width: 100%;
      }
    `;
  },
  sx,
  system,
);

export const Embed = React.forwardRef<HTMLDivElement, EmbedProps>(({ children, ...rest }, ref) => (
  <StyledEmbed ref={ref} dangerouslySetInnerHTML={{ __html: children }} {...rest} />
));

Embed.defaultProps = {
  ratio: 9 / 16,
};
