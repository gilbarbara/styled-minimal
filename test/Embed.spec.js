import React from 'react';
import { mount } from 'enzyme';

import Embed from '../src/Embed';

describe('Embed', () => {
  const wrapper = mount(
    <Embed>
      {`<iframe
          width="560"
          height="315"
          src="https://www.youtube.com/embed/GNCd_ERZvZM"
          title="Them Changes"
          frameBorder="0"
          allowFullScreen
        />`}
    </Embed>,
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
