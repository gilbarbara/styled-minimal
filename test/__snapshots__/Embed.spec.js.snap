// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Embed should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  height: 0;
  padding-bottom: 56.25%;
  overflow: hidden;
  position: relative;
  width: 100%;
}

.c0 > iframe {
  border: 0;
  bottom: 0;
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
}

<StyledEmbed
  dangerouslySetInnerHTML={
    Object {
      "__html": "<iframe
          width=\\"560\\"
          height=\\"315\\"
          src=\\"https://www.youtube.com/embed/GNCd_ERZvZM\\"
          title=\\"Them Changes\\"
          frameBorder=\\"0\\"
          allowFullScreen
        />",
    }
  }
  ratio={0.5625}
>
  <div
    className="c0"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<iframe
          width=\\"560\\"
          height=\\"315\\"
          src=\\"https://www.youtube.com/embed/GNCd_ERZvZM\\"
          title=\\"Them Changes\\"
          frameBorder=\\"0\\"
          allowFullScreen
        />",
      }
    }
  />
</StyledEmbed>
`;
